File: 0055-jump-game/0055-jump-game.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    let goal = nums.length - 1
    for(let i = nums.length - 1; i >= 0; i--){ // start from the end and move the goal closer
        if(i + nums[i] >= goal){
            goal = i
        }
    }

    return goal === 0 ? true : false  
};

File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        goal_idx = len(nums)-1
        
        for i in range(len(nums)-2, -1, -1):
            if nums[i] + i >= goal_idx:
                goal_idx = i
        
        return goal_idx == 0